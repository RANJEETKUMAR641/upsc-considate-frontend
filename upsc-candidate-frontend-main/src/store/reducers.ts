/* eslint-disable */

import { combineReducers } from 'redux'
import reducers from 'app/components/reducers'
import { connectRouter } from 'connected-react-router'

import history from 'utils/history'

import { InjectedReducersType } from 'utils/types/injector-typings'

export default function createReducer(
  injectedReducers: InjectedReducersType = {},
) {
  return combineReducers({
    ...injectedReducers,
    ...reducers,
    router: connectRouter(history),
  })
}
