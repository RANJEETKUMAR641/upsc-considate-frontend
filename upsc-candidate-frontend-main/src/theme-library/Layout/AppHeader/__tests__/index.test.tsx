import { render, screen } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { configureAppStore } from 'store/configureStore'

import { Provider } from 'react-redux'
import Header from '../index'

jest.mock('axios')

jest.mock('')
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    }
  },
}))

describe('Test cases for login page header snapshot', () => {
  const queryClient = new QueryClient()
  const store = configureAppStore()

  it('should match snapshot', () => {
    const loadingIndicator = render(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <Header />
        </QueryClientProvider>
      </Provider>,
    )
    expect(loadingIndicator.container.firstChild).toMatchSnapshot()
  })
})

describe('Test cases for element present', () => {
  it('should element present in document', () => {
    const queryClient = new QueryClient()
    const store = configureAppStore()

    render(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <Header />
        </QueryClientProvider>
      </Provider>,
    )
    const text = screen.getByTestId('logoTxt')
    const element = screen.getByTestId('OTR')
    expect(text).toBeInTheDocument()
    expect(element).toBeInTheDocument()
  })
})
