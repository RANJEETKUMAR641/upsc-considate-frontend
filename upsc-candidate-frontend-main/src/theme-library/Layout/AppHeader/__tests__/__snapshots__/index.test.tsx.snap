// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for login page header snapshot should match snapshot 1`] = `
<div>
  <div
    aria-label="Site Header"
    class="app-header header-shadow appear appear-active"
    component="div"
  >
    <div
      class="app-header__logo"
    >
      <div
        class="sc-bczRLJ kxxrZ logoTxt"
      >
        <img
          alt="Logo"
          src="IMAGE_MOCK"
        />
        <h1
          class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
        >
          संघ लोक सेवा आयोग
          <p>
            UNION PUBLIC SERVICE COMMISSION
          </p>
        </h1>
      </div>
    </div>
    <div
      class="app-header__mobile-menu"
    >
      <div
        role="button"
        tabindex="0"
      >
        <button
          class="jsx-3928250682 BurgerSlider Burger"
        >
          <div
            class="jsx-3928250682 BurgerBox"
          >
            <div
              class="jsx-3928250682 BurgerInner"
            />
          </div>
        </button>
      </div>
    </div>
    <div
      class="app-header__menu"
    >
      <span
        role="button"
        tabindex="0"
      >
        <button
          class="btn-icon btn-icon-only btn btn-dark btn-sm"
          type="button"
        >
          <div
            class="btn-icon-wrapper"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-ellipsis-v "
              data-icon="ellipsis-v"
              data-prefix="fas"
              focusable="false"
              role="img"
              style="vertical-align: 0rem;"
              viewBox="0 0 192 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                fill="currentColor"
              />
            </svg>
          </div>
        </button>
      </span>
    </div>
    <div
      class="app-header__content"
    >
      <div
        class="app-header-right"
      />
    </div>
  </div>
</div>
`;
